sympy/solvers/bivariate.py

```diff
@@ -199,17 +244,61 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in list(set(flatten(power_list(lhs, symbol)))) \
+                    if i%2 == 0 and i != 0]
+    t = Dummy('t', **symbol.assumptions0)
+
     lamcheck = [tmp for tmp in gens
                 if (tmp.func in [exp, log] or
                 (tmp.is_Pow and symbol in tmp.exp.free_symbols))]
     if not lamcheck:
         raise NotImplementedError()
 
+    def _solve_even_degree_branches(diff, symbol, t, gens):
```

Comment:
1. The name "even degree branches" seems a bit misleading. For Lambert we are considering only two branches 0 and -1. Maybe `_solve_even_degree_expr` will be better. Also clearly mention in the documentation what does it mean and what it does.

2. I think you can access `t` within the routine, so no need to pass it as parameter. Same goes with `gens`.

3. Also follow the conventional layout of writing documentation (like in solveset).

4. `_solve_...(expr,  symbol)` will be better?

Reply:
Done this